docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
+  managed:
+    match:
+      type: KumaHostname # Kuma will generate a domain
+      port: 80
+      protocol: http
+    extension:
+      type: Lambda 
+      config: # type JSON
+        arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+    endpoints:
+      - address: 1.1.1.1
+        port: 12345
+      - address: example.com
+      - address: unix://....
+    tls:
+      version:
+        min: TLSv1_2 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+        max: TLSv1_3 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+      allowRenegotiation: false
+      verification:
+        skipSAN: true # if this is true then subjectAltNames don't take effect
+        subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+          - example.com
+          - "spiffe://example.local/ns/local"
+        caCert: 
+          inline: 123
+        clientCert:
+          secret: 123
+        clientKey:
+          secret: 123
+status: 
+  vip:
+    value: 242.0.0.1
+    type: Kuma
+  addresses:
+  - hostname: example.ext.svc.local
+    status: NotAvailable
+    origin:
+      kind: HostnameGenerator
+      name: k8s-example-service-hostname
+    reason: "addresses are overlapping with ext2"
+```
+* **spec**:
+  * **match**: defines a traffic that should be routed through the sidecar
+    * **type**: type of the match, one of `KumaHostname`, `Domain`, `CIDR` and `IP` are available
+      * `KumaHostname`: allocates a VIP for a domain provided in the `value` field
+      * `Domain`: handles traffic to the specified domain
+      * `CIDR`: handles traffic to specified addresses range
+      * `IP` handles the traffic to specified IP
+    * **value**: depends of the type can be an existing domain, new domain name, CIDR or IP
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tls`: should be used when TLS traffic is originated by the client application in the case the `kuma.io/protocol` would be tcp
+      * `tcp`: WARNING: shouldn't be used when match has only domains. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port. We are going to validate configuration and do not apply config when protocol is tcp and `type: Domain`.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **type**: defines what kind of destination it is, one of `Managed`, `Passthrough`, or `Extension`, (Default: `Passthrough`)
+    * `Managed`: allows creating a set of destination endpoints and `TLS` configuration, when defined section `extension` is not available.
+  * **managed**: defines where matched requests should be routed, only for a `type: Managed`
+    * **extension**: struct for a plugin configuration, only for a `type: Extension`
```

Comment:
If it's `only for a type: Extension`, shouldn't this field be placed on the same level as `managed`?

Reply:
I think it missed this. I cleaned up